src/lib/navigation/navigation-mode.interface.ts

```diff
@@ -1,64 +1,81 @@
-import {WizardState} from './wizard-state.model';
 import {EventEmitter} from '@angular/core';
 
 /**
- * An interface describing the basic functionality, which must be provided by a navigation mode.
+ * An interface containing the basic functionality, which must be provided by a navigation mode.
  * A navigation mode manages the navigation between different wizard steps, this contains the validation, if a step transition can be done
  *
+ * For base implementation see [[BaseNavigationMode]].
+ *
  * @author Marc Arndt
  */
-export abstract class NavigationMode {
-  constructor(protected wizardState: WizardState) {
-  }
+export interface NavigationMode {
```

Comment:
I think the comments to the methods in the interface should be reviewed.
For example the comment for `goToStep` contains:

https://github.com/madoar/angular-archwizard/blob/74c0050a6b41722fd0a138fa70d110067e8bf560/src/lib/navigation/navigation-mode.interface.ts#L34-L39

The issue is: Does this apply for all possible navigation modes? I think it is more fitting to move such "constraints" to the `BaseNavigationMode` class, because it specifies our "base" behavior to the content described in the comment.

Reply:
I completely agree

Reply:
Moved a bunch of comments to `BaseNavigationMode` in b4a5601